R/geom_text_lastonly.R

```diff
@@ -1,310 +1,322 @@
+#'   TRUE, add_points = FALSE, text_aes = NULL, point_aes = NULL, ...)
+#'
+#' @inheritParams ggplot2::layer
+#' @inheritParams ggplot2::geom_point
+#' @param parse If \code{TRUE}, the labels will be parsed into expressions and
+#'   displayed as described in \code{?plotmath}.
+#' @param nudge_x,nudge_y Horizontal and vertical adjustment to nudge labels by.
+#'   Useful for offsetting text from points, particularly on discrete scales.
+#'   Cannot be jointy specified with \code{position}.
+#' @param position Position adjustment, either as a string, or the result of a
+#'   call to a position adjustment function. Cannot be jointy specified with
+#'   \code{nudge_x} or \code{nudge_y}.
+#' @param check_overlap If \code{TRUE}, text that overlaps previous text in the
+#'   same layer will not be plotted. \code{check_overlap} happens at draw time
+#'   and in the order of the data. Therefore data should be arranged by the
+#'   label column before calling \code{geom_text_lastonly()}.
+#' @param add_points If \code{TRUE}, points will be added to the plot (for the
+#'   labeled data only). Default size=2, color will match line color.
+#' @param text_aes,point_aes Named list, additional aesthetics to send to the
+#'   text and point geoms, respectively.
+#' @param ... additional aesthetics to send to BOTH the point and text geoms.
+#'   Note that if \code{add_points = FALSE}, additional parameters can be passed
+#'   to the text geom here, rather than in \code{text_aes}, without breaking.
+#'
+#' @examples
+#' df <- data.frame(year=2010:2020, value=runif(22), var=c(rep("A", 11), rep("B", 11)))
+#'
+#' # Without points, label formatting or x-axis expansion
+#' ggplot(df, aes(x=year, y=value, color=var)) +
+#'   geom_line() +
+#'   labs(title="Random lines") +
+#'   scale_y_continuous("Percentage of absolutely nothing") +
+#'   scale_x_continuous("Year") +
+#'   geom_text_lastonly()
+#'
+#' # With points, label formatting and x-axis expansion
+#' ggplot(df, aes(x=year, y=value, color=var, label=sprintf("%.1f%%", 100*var))) +
+#'   geom_line() +
+#'   labs(title="Random lines") +
+#'   scale_y_continuous("Percentage of absolutely nothing", labels=scales::percent) +
+#'   scale_x_continuous("Year", expand=expansion(mult=c(0.05, 0.10))) +
+#'   geom_text_lastonly(add_points=TRUE, text_aes=list(fontface="bold"), point_aes=list(size=2.5))
+#'
+#' @export
+geom_text_lastonly <- function(mapping = NULL, data = NULL,
+                      stat = "identity", position = NULL,
+                      parse = FALSE,
+                      nudge_x = 0.25,
+                      nudge_y = 0,
+                      check_overlap = FALSE,
+                      na.rm = FALSE,
+                      show.legend = FALSE,
+                      inherit.aes = TRUE,
+                      add_points = FALSE,
+                      text_aes = NULL,
+                      point_aes = NULL,
+                      ...)

```

Comment:
I moved `...` to last position in this file and in `geom_recessions`. I can't imagine this is too big a deal but flagging for Noel in case you have concerns.

Reply:
The positioning for all of the `...` parameters found in `geom_text_lastonly` was identical to the positioning in the original `ggplot` code that I copied. I personally would be inclined to leave them where they were before, but I won't put up a fight over it.

Reply:
OK, I am reverting.